x/programs/simulator/state/state.go

```diff
@@ -0,0 +1,172 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package state
+
+/*
+#cgo CFLAGS: -I../common
+#include "callbacks.h"
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"context"
+	"fmt"
+	"unsafe"
+
+	"github.com/ava-labs/avalanchego/database"
+
+	"github.com/ava-labs/hypersdk/codec"
+	"github.com/ava-labs/hypersdk/state"
+)
+
+// ensure SimulatorState implements state.Mutable
+var _ state.Mutable = &SimulatorState{}
+
+type SimulatorState struct {
+	// this is a pointer to the state passed in from rust
+	// it points to the state object on the rust side of the simulator
+	statePtr unsafe.Pointer
+
+	// this is ptr to the get function
+	getFunc C.GetStateCallback
+
+	// this is ptr to the insert function
+	insertFunc C.InsertStateCallback
+
+	// this is a ptr to the delete function
+	removeFunc C.RemoveStateCallback
+}
```

Comment:
I wonder if we can define this in `types.h` and have `types.h` depend on `callbacks.h`? Or maybe you have to import both?

Reply:
Oh... but I guess that would mean there's a circular dependency? 

Seems like there should be a better way to define a shared version of the `Mutable` struct.